code/game/dna/mutations/powers.dm

```diff
@@ -1179,3 +1157,54 @@
 	else
 		H.remoteview_target = null
 		H.reset_perspective()
+
+/datum/mutation/meson_vision
+	name = "Meson Vision"
+	activation_messages = list("More information seems to reach your eyes...")
+	deactivation_messages = list("The amount of information reaching your eyes fades...")
+	instability = GENE_INSTABILITY_MINOR
+	traits_to_add = list(TRAIT_MESON_VISION)
+
+/datum/mutation/meson_vision/New()
+	..()
+	block = GLOB.mesonblock
+
+/datum/mutation/meson_vision/activate(mob/living/M)
+	..()
+	M.update_sight()
+
+/datum/mutation/meson_vision/deactivate(mob/living/M)
+	..()
+	M.update_sight()
+
+/datum/mutation/night_vision
+	name = "Night Vision"
+	activation_messages = list("Were the lights always that bright?")
+	deactivation_messages = list("The ambient light level returns to normal...")
+	instability = GENE_INSTABILITY_MODERATE
+	traits_to_add = list(TRAIT_NIGHT_VISION)
+
+/datum/mutation/night_vision/New()
+	..()
+	block = GLOB.nightvisionblock
+
+/datum/mutation/night_vision/activate(mob/living/M)
+	..()
+	M.update_sight()
+	M.update_icons()
+
+/datum/mutation/night_vision/deactivate(mob/living/M)
+	..()
+	M.update_sight()
+	M.update_icons()
```

Comment:
I could be missing something here, but as far as I can tell nothing gets changed in `update_icons()`.
```suggestion
/datum/mutation/night_vision/activate(mob/living/M)
	..()
	M.update_sight()

/datum/mutation/night_vision/deactivate(mob/living/M)
	..()
	M.update_sight()
```

Reply:
Mobs with night vision have glowing eyes

Reply:
Ah, true. `update_icons()` does seem to be the wrong proc to modify that though, it doesn't have any human overrides:
![image](https://user-images.githubusercontent.com/57483089/127360121-6931e1c8-4f09-4936-901b-f6eee2daef37.png)
Night vision causing eye shine seems to be set here:
https://github.com/ParadiseSS13/Paradise/blob/fba7585b4ea094c3aa91547d9bbb8caadfc69962/code/modules/surgery/organs/eyes.dm#L40-L42
Which in turn is called by `eyes_shine()`:
https://github.com/ParadiseSS13/Paradise/blob/fba7585b4ea094c3aa91547d9bbb8caadfc69962/code/modules/mob/living/carbon/human/human.dm#L1487-L1496
Which is used to decide if an overlay should be added in `update_misc_effects()`:
https://github.com/ParadiseSS13/Paradise/blob/fba7585b4ea094c3aa91547d9bbb8caadfc69962/code/modules/mob/living/carbon/human/update_icons.dm#L1271-L1278

So you probably want this instead, for the night vision gene:
```suggestion
/datum/mutation/night_vision/activate(mob/living/M)
	..()
	M.update_sight()
	if(ishuman(M))
		var/mob/living/carbon/human/H = M
		H.update_misc_effects()

/datum/mutation/night_vision/deactivate(mob/living/M)
	..()
	M.update_sight()
	if(ishuman(M))
		var/mob/living/carbon/human/H = M
		H.update_misc_effects()
```